src/msg/async/AsyncConnection.h

```diff
@@ -104,8 +104,16 @@ class AsyncConnection : public Connection {
     return !out_q.empty() || outcoming_bl.length();
   }
   void shutdown_socket() {
-    if (sd >= 0)
+    for (auto &&t : register_time_events)
```

Comment:
`auto t` would suffice. there is no need to use universal reference here. and the primitive type would be clearer.